app/models/movie.rb

```diff
@@ -1,5 +1,11 @@
 class Movie < ApplicationRecord
   has_many :actor_movies
   has_many :actors, through: :actor_movies
+  has_many :movie_genres
+  has_many :genres, through: :movie_genres
   belongs_to :director
+
+  def self.average_rating
```

Comment:
you don't want this to be a class method, nor in the movie class, it'd be better to have this as an instance method for genre, so you could do
```ruby
class MovieGenre ...

  def movie_average
     movies.average(:rating)
  end
end
```

Reply:
otherwise this would pull the average for ALL movies (because it's a class method)